src/docdb/getCosmosClient.ts

```diff
@@ -4,15 +4,70 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { CosmosClient } from "@azure/cosmos";
+import { getConfiguredAuthProviderId } from "@microsoft/vscode-azext-azureauth";
 import { appendExtensionUserAgent } from "@microsoft/vscode-azext-utils";
 import * as https from "https";
 import * as vscode from 'vscode';
 import { ext } from "../extensionVariables";
 
-export function getCosmosClient(endpoint: string, key: string, isEmulator: boolean | undefined): CosmosClient {
+export type CosmosDBKeyCredential = {
+    type: "key";
+    key: string;
+};
+
+export type CosmosDBAuthCredential = {
+    type: "auth";
+};
+
+export type CosmosDBCredential = CosmosDBKeyCredential | CosmosDBAuthCredential;
+
+/**
+ * Duplicated from @microsoft/vscode-azext-azureauth
+ * @todo: Use the subscription client once it supports customizing scopes.
+ */
+export async function getSessionFromVSCode(scopes: string | string[], options?: vscode.AuthenticationGetSessionOptions): Promise<vscode.AuthenticationSession | undefined> {
+    const scopesArray = typeof scopes === "string" ? [scopes] : scopes;
+    return await vscode.authentication.getSession(getConfiguredAuthProviderId(), scopesArray, options);
+}
+
+export function getCosmosClient(
+    endpoint: string,
+    credentials: CosmosDBCredential[],
+    isEmulator: boolean | undefined
+): CosmosClient {
     const vscodeStrictSSL: boolean | undefined = vscode.workspace.getConfiguration().get<boolean>(ext.settingsKeys.vsCode.proxyStrictSSL);
     const enableEndpointDiscovery: boolean | undefined = vscode.workspace.getConfiguration().get<boolean>(ext.settingsKeys.enableEndpointDiscovery);
     const connectionPolicy = { enableEndpointDiscovery: (enableEndpointDiscovery === undefined) ? true : enableEndpointDiscovery };
-    return new CosmosClient({ endpoint, key, userAgentSuffix: appendExtensionUserAgent(), agent: new https.Agent({ rejectUnauthorized: isEmulator ? !isEmulator : vscodeStrictSSL }), connectionPolicy: connectionPolicy });
 
+    const keyCred = credentials.filter((cred): cred is CosmosDBKeyCredential => cred.type === "key")[0];
```

Comment:
🙃

Reply:
The 'lib' in tsconfig is too old to support .at. Changing it results in multiple build error to surface. I'll skip using .at in this PR.